xml/ha_config_basics.xml

```diff
@@ -200,7 +408,88 @@
+   # see also corosync.conf.5 and votequorum.5
+   provider: corosync_votequorum
+   expected_votes: 2
+   two_node: 1
+}</screen>
+   </example>
+   <para>
+    Normally you do not need to change this. Do not set
+    <literal>no-quorum-policy=ignore</literal> and <literal>two_node: 1</literal>
+    together. By default, when <literal>two_node: 1</literal> is set, the
+    <literal>wait_for_all</literal> option is automatically enabled.
+    If <literal>wait_for_all</literal> is not enbaled, the cluster should be
+    started on both nodes in parallel. Otherwise the first node will perform
+    a startup-fencing on the missing second node.
+   </para>
+  </sect2>
+  <sect2 xml:id="sec.ha.config.basics.corosync.n-node">
+   <title>&corosync; Configuration for N-Node Clusters</title>
+   <para>
+    When not using a two-node cluster, we strongly recommend an odd number
+    of nodes four your N-node cluster. With regards to quorum configuration,
+    you have the following options:
+   </para>
+   <itemizedlist>
+    <listitem>
+     <para>Adding additional nodes with the <command>ha-cluster-join</command>
+     command, or</para>
+    </listitem>
+    <listitem>
+     <para>Adapting the &corosync; configuration manually.</para>
+    </listitem>
+   </itemizedlist>
+   <para>
+    If you adjust <filename>/etc/corosync/corosync.conf</filename> manually,
+    use the following settings:
+   </para>
+   <example>
+    <title>Excerpt of &corosync; Configuration for a N-Node Cluster</title>
+    <screen>quorum {
+   provider: corosync_votequorum <co xml:id="co.corosync.quorum.n-node.corosync_votequorum"/>
+   expected_votes: <replaceable>N</replaceable> <co xml:id="co.corosync.quorum.n-node.expected_votes"/>
+   wait_for_all: 1 <co xml:id="co.corosync.quorum.n-node.wait_for_all"/>
+}</screen>
+    <calloutlist>
+     <callout arearefs="co.corosync.quorum.n-node.corosync_votequorum">
+      <para>Use the quorum service from &corosync;</para>
+     </callout>
+     <callout arearefs="co.corosync.quorum.n-node.expected_votes">
+      <para>The number of votes to expect. This parameter can either be
+       provided inside the <literal>quorum</literal> section, or is
+       automatically calculated when the <literal>nodelist</literal>
+       section is available.</para>
+     </callout>
+     <callout arearefs="co.corosync.quorum.n-node.wait_for_all">
+      <para>
+       Enables the wait for all (WFA) feature.
+       When WFA is enabled, the cluster will be quorate for the first time
+       only after all nodes have been visible.
+       To avoid some startup race conditions, setting <option>wait_for_all</option>
+       to <literal>1</literal> may help.
+       For example, in a five-node cluster every node has one vote and thus,
+       <option>expected_votes</option> is set to <literal>5</literal>.
+       As soon as three or more nodes are visible to each other, the cluster
+       partition becomes quorate and can start operating.
```

Comment:
This actually is the behavior with the default "wait_for_all: 0". 

Or probably we can describe the default behavior first and then for "wait_for_all: 1"?

Reply:
And be aware that "wait_for_all" defaults to "0". In the previous 2-node scenario, it's just "two_node_: 1" automatically enables it.

Reply:
@gao-yan Hmn...

> Or probably we could describe the default behavior first and then for "wait_for_all: 1"?

But then, I can't describe both in the section "Configuration for N-Node Clusters", right? I'm not sure how to integrate it. Any ideas?

Reply:
Hrm, I've been thinking about this...

Since we first mention "wait_for_all" in 2-Node scenario, maybe we could already start introducing the option in details there like:

```
When "two_node: 1" is set, the wait_for_all option is automatically enabled if it's not explicitly disabled. 

When wait_for_all is enabled, cluster will be quorate for the first time only after all nodes have been visible. Hence to avoid some startup race conditions, keeping wait_for_all enabled may help.

Depending on the needs,  wait_for_all could be explicitly set to be disabled even with "two_node:1". In that case, on cluster startup,  even if only one node is visible,  the cluster partition becomes quorate and can start operating. Hence it will perform startup-fencing on the missing second node.
```
In N-Node scenario, it makes more sense to just keep "wait_for_all" default. So probably we should just remove the setting "wait_for_all: 1" from the configuration example, any say like:
```
wait_for_all defaults to 0 in this case.  So for example, in a five-node cluster every node has one vote and thus, "expected_votes" is set to "5".  As soon as three or more nodes are visible to each other, the cluster partition becomes quorate and can start operating.
```
I think "wait_for_all: 1" has been described well enough in above, so no need to repeat I guess.